src/librustdoc/clean/utils.rs

```diff
@@ -33,11 +34,11 @@ crate fn krate(cx: &mut DocContext<'_>) -> Crate {
 
     let mut externs = Vec::new();
     for &cnum in cx.tcx.crates(()).iter() {
-        externs.push((cnum, cnum.clean(cx)));
+        externs.push(ExternalCrate { crate_num: cnum });
         // Analyze doc-reachability for extern items
         LibEmbargoVisitor::new(cx).visit_lib(cnum);
     }
-    externs.sort_by(|&(a, _), &(b, _)| a.cmp(&b));
+    externs.sort_unstable();
```

Comment:
I'm wondering if maybe it would be better to `sort_unstable_by_key(|c| c.crate_num)` instead (and then remove the `Ord` derive from `ExternalCrate`), to make it more explicit what the crates are being sorted by. Also, it would the code more resilient to change in `ExternalCrate`. But I'm okay with leaving it how it is now if that's what you'd prefer. What do you think?

Reply:
Actually, on second thought, why do we sort the `externs` at all? Is it to prevent non-determinism? Can you try switching it to use `BTreeSet` instead?

Reply:
The order is used here: https://github.com/rust-lang/rust/blob/23fdc18eb68b1ec5daee2d095441d47afb0ce89e/src/librustdoc/formats/cache.rs#L161-L165
I have no idea why, the comment doesn't make much sense to me since the crate number has very little relation to the dependency depth it's at. I would be fine with removing it.

Reply:
Oh! This would be a *great* way to handle https://github.com/rust-lang/rust/issues/73423, we can have different docs in core than in std. I didn't realize rustdoc silently overwrote the new docs.

Reply:
It turns out the sort can't be removed :/ it would cause the docs in core to take precedence over those defined in downstream crates. That wouldn't be a problem, except that `doc(primitive)` is stable for some reason and it would be a change in observable behavior.

I'll add back the sorting. I still think this is the right approach for fixing 73423 though.